File: 인수경/230115/pg_구명보트.js
function solution(A, K) {
  let count = 0;
  A.sort((a, b) => a - b);

  while (A.length > 0) {
    for (let i = A.length - 1; i >= 0; i--) {
      if (A[0] + A[i] <= K) {
        count++;
        A.splice(i, 1);
        A.splice(0, 1);
        break;
      } else {
        count++;
        A.splice(i, 1);
      }
    }
  }
  return count;
}

File: 인수경/230115/pg_구명보트.py
def solution(people, limit):
    answer = 0
    people.sort(reverse=True)

    top = 0
    rear = len(people) - 1

    while top <= rear:
        if top == rear:
            answer += 1
            break

        if people[top] + people[rear] > limit:
            top += 1
            answer += 1
        else:
            top += 1
            rear -= 1
            answer += 1

    return answer
